examples_tests/61.BlitFilterTest/default_compute_normalization.comp

```diff
@@ -0,0 +1,192 @@
+	#endif
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 2)
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ uimage2D
+	#else
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ image2D
+	#endif
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 3)
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ uimage3D
+	#else
+		#define _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_TYPE_ image3D
+	#endif
+#else
+	#error _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_ not supported
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_PIXEL_TYPE_DEFINED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_PIXEL_TYPE_DEFINED_
+struct nbl_glsl_blit_normalization_pixel_t
+{
+	vec4 data;
+};
+#endif
+
+#define _NBL_GLSL_WORKGROUP_SIZE_ (_NBL_GLSL_WORKGROUP_SIZE_X_ * _NBL_GLSL_WORKGROUP_SIZE_Y_ * _NBL_GLSL_WORKGROUP_SIZE_Z_)
+
+layout (local_size_x = _NBL_GLSL_WORKGROUP_SIZE_X_, local_size_y = _NBL_GLSL_WORKGROUP_SIZE_Y_, local_size_z = _NBL_GLSL_WORKGROUP_SIZE_Z_) in;
+
+#include <../normalization/parameters.glsl>
+#include <../normalization/normalization.glsl>
+#include <../normalization/descriptors.glsl>
+// #include <../blit/formats/rgba16.glsl>
+
+layout (push_constant) uniform Block
+{
+	nbl_glsl_blit_normalization_parameters_t params;
+} pc;
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_GET_PARAMETERS_DEFINED_
+nbl_glsl_blit_normalization_parameters_t nbl_glsl_blit_normalization_getParameters()
+{
+	return pc.params;
+}
+#define _NBL_GLSL_BLIT_NORMALIZATION_GET_PARAMETERS_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_GET_PADDED_DATA_DEFINED_
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_
+#error _NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_ must be defined
+#endif
+
+nbl_glsl_blit_normalization_pixel_t nbl_glsl_blit_normalization_getPaddedData(in ivec3 coord)
+{
+	const nbl_glsl_blit_normalization_parameters_t params = nbl_glsl_blit_normalization_getParameters();
+
+#if NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 1)
+	#define COORD coord.x
+	#define IMAGE_DIM params.outImageDim.x
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 2)
+	#define COORD coord.xy
+	#define IMAGE_DIM params.outImageDim.xy
+#elif NBL_GLSL_EQUAL(_NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_, 3)
+	#define COORD coord
+	#define IMAGE_DIM params.outImageDim
+#else
+	#error _NBL_GLSL_BLIT_NORMALIZATION_DIM_COUNT_ not supported
+#endif
+
+	nbl_glsl_blit_normalization_pixel_t result;
+	result.data = vec4(0.f);
+	if (all(lessThan(COORD, IMAGE_DIM)))
+	{
+	#ifdef _NBL_GLSL_BLIT_NORMALIZATION_SOFTWARE_CODEC_
+		result.data = nbl_glsl_blit_formats_decode(imageLoad(_NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_, COORD));
+	#else
+		result.data = imageLoad(_NBL_GLSL_BLIT_NORMALIZATION_INOUT_IMAGE_DESCRIPTOR_DEFINED_, COORD);
+	#endif
+	}
```

Comment:
because your input (intermediate) image will need to be higher precision than output, you might as well use a sample-able format, so then you can just do `texelFetch` and not worry about format decode

Reply:
Resolved.